src/debug-integration.adoc

```diff
@@ -107,3 +110,22 @@ The <<dscratch1c>> register is a CLEN-bit plus tag bit extension to
 
 .Debug scratch 1 capability register
 include::img/dscratch1creg.edn[]
+
+[#dinfc,reftext="dinfc"]
+==== Debug Infinite Capability Register (dinfc)
+
+The <<dinfc>> register is a CLEN-bit plus tag bit CSR only accessible in debug mode.
+
+The reset value is the <<infinite-cap>> capability. The M field (if
+{cheri_mode_ext_name} is implemented) represents the current mode and
+will be only updated if <<MODESW>> is executed.
+
+<<dinfc>> is read/write but with no writeable fields, and so writes are
+ignored.
+
+NOTE: A future version of this specification may add writeable fields to allow creation
```

Comment:
If there are multiple roots in the future couldn't we just add new debug-"infinite code cap, infinite data cap" CSRs and have this one be one of them? Writing to this CSR seems a bit odd since it's then stateful and we have to keep track of that state while in debug mode?

Reply:
The issue is there's only space for 16 debug-only CSRs and with this change 14 of them are used.

Reply:
dcsr, dpc, dscratch0, dscratch1, dddc, dpcc, dscratch0c, dscratch1c, what are the 5 I'm missing?

Reply:
One identified problem here was if we introduced an otype later, or something else, for compartmentalization. Would we need to redefine this CSR? The most basic idea was to have this as a CSR where the tag is _always_ set and so the software can write whatever it likes and have it tagged. This idea then extended to different root, or anything else that we may need in the future.

The other reason not to add more CSRs is that there are only 16 available. There are four standard ones: 
<img width="463" alt="image" src="https://github.com/riscv/riscv-cheri/assets/59061376/86ba7473-5303-4afe-b15e-796d02a2b201">

and we've added an alias to three of them, so that's 7. 
We've also added `dddc` and `dinfc`, making 9 so we're quite close to filling the whole space, and then there's no provision for adding any more without some kind of banking scheme.

So I'd really like to get the "make any cap" functionality based on a single debug CSR if possible.